File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[]}
     */
    inorderTraversal(root) {
        let res = [];

        function inorder(node) {
            if (node === null) {
                return;
            }

            inorder(node.left)
            res.push(node.val);
            inorder(node.right);
        }

        inorder(root);

        return res;
    }
}

File: Data Structures & Algorithms/same-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        if not p and not q: 
            return True
        
        if q and p and q.val == p.val:
            return self.isSameTree(q.left, p.left) and self.isSameTree(p.right, q.right)
        else:
            return False